core/string.c

```diff
@@ -117,28 +117,36 @@ d_r_strrchr(const char *str, int c)
 /* Private strncpy.  Standard caveat about not copying trailing null byte on
  * truncation applies.
  */
+#define D_R_STRNCPY_BODY()                        \
+    do {                                          \
+        size_t i;                                 \
+        for (i = 0; i < n && src[i] != '\0'; i++) \
+            dst[i] = src[i];                      \
+        /* Pad the rest with nulls. */            \
+        for (; i < n; i++)                        \
+            dst[i] = '\0';                        \
+        return dst;                               \
+    } while (0);
 char *
 d_r_strncpy(char *dst, const char *src, size_t n)
 {
-    size_t i;
-    for (i = 0; i < n && src[i] != '\0'; i++)
-        dst[i] = src[i];
-    /* Pad the rest with nulls. */
-    for (; i < n; i++)
-        dst[i] = '\0';
-    return dst;
+    D_R_STRNCPY_BODY()
```

Comment:
style: End in a semicolon here.

Reply:
(This is not just a DR arbitrary style choice: hiding the ; messes up parsers and hinders readability.)

Reply:
Fixed.